sqlgen/dialect_pgx.go

```diff
@@ -4,11 +4,26 @@
 
 package sqlgen
 
-import "strconv"
+import (
+	"github.com/jackc/pgx/v5/pgtype"
+	"strconv"
+)
 
 // this type is specially named to match up with the name returned by the
 // dialect impl in the sql package.
-type pgx struct{}
+type pgx struct {
+	scanMap *pgtype.Map
```

Comment:
One of the drawbacks at the moment is that this will be included despite the actual implementation included. I'm not sure whether it ends up in the final binary when you are using only one backend (e.g. sqlite). However, it's not a critical issue per se.

Reply:
Yes pgtype does end up in there even with sqlite. This is a drawback.

I could move this to a template file. Another approach would be make the sqlgen/sqlbundle aware of dialects and only include the requested ones.

It doesn't seem to get any easier with either of those approaches.